Update Documentation for Places Endpoint

ehunter-usgs opened the pull request:
fixes usgs/hazdev-geoserve-ws#45

jmfee-usgs commented:
- add `id` attributes for linking to specific parameters
- use description list?
- option/type could be part of description, to make parameters easier to scan.

ehunter-usgs commented:
- Added id attributes for linking to parameters
 - Updated markup to be more semantic, but still using lists (for layout)
 - kept option/type where it is, this is supposed to be easier to scan for "required" parameters.

jmfee-usgs commented:
- Make anchors use the same name as parameter
- maxradiuskm has incorrect anchor
- pretty sure event type doesn't work with rectangle
- separate circle/rectangle from shared parameters, note differences within parameter
